### PR TITLE
Unacknowledged Opcode based writes aren't supported

### PR DESCRIPTION
**Pull request by rozza**
JAVA-2241

### CODE REVIEW

File: driver-core/src/main/com/mongodb/operation/MixedBulkWriteOperation.java [hunk 1]
```diff
@@ -170,12 +168,8 @@ public BulkWriteResult execute(final WriteBinding binding) {
         return withConnection(binding, new CallableWithConnection<BulkWriteResult>() {
             @Override
             public BulkWriteResult call(final Connection connection) {
-                if (bypassDocumentValidationNotSupported(bypassDocumentValidation, writeConcern, connection.getDescription())) {
-                    throw getBypassDocumentValidationException();
-                }
-
-                checkValidWriteRequestCollations(connection, writeRequests);
-
+                checkValidBypassDocumentValidationAndWriteRequestCollations(connection, bypassDocumentValidation, writeRequests,
```
Comment (jyemin) [inline driver-core/src/main/com/mongodb/operation/MixedBulkWriteOperation.java]: These names are going to get even longer if we add a third thing to check for.  How about just

    validateWriteRequests

?

**Comment by jyemin** [comment]
Except for the suggestion to simplify some of the method names, it looks good.

**Comment by rozza** [comment]
Updated the method name to match the convention.

Then in a separate commit updated the convention.

**Comment by jyemin** [comment]
LGTM

**Pull request closed without merging**